api/app/utils/cffdrs.py

```diff
@@ -296,11 +401,120 @@ def get_ffmc_for_target_hfi(fuel_type: str,
             experimental_ffmc = min(101, experimental_ffmc + ((101 - experimental_ffmc)/2))
         else:  # if the error value is a negative number, need to make experimental FFMC value smaller
             experimental_ffmc = max(0, experimental_ffmc - ((101 - experimental_ffmc)/2))
+        experimental_isi = initial_spread_index(experimental_ffmc, wind_speed)
         experimental_sfc = surface_fuel_consumption(fuel_type, bui, experimental_ffmc, percentage_conifer)
+        experimental_ros = rate_of_spread(fuel_type, experimental_isi, bui, fmc,
+                                          experimental_sfc, percentage_conifer,
+                                          grass_cure, percentage_dead_balsam_fir, crown_base_height)
         experimental_hfi = head_fire_intensity(fuel_type,
                                                percentage_conifer,
                                                percentage_dead_balsam_fir,
-                                               bui, experimental_ffmc, ros, cfb, cfl, experimental_sfc)
+                                               bui, experimental_ffmc, experimental_ros,
+                                               cfb, cfl, experimental_sfc)
         error_hfi = (target_hfi - experimental_hfi) / target_hfi
 
     return (experimental_ffmc, experimental_hfi)
+
+
+def get_critical_hours_start(critical_ffmc: float, solar_noon_ffmc: float, temperature: float,
```

Comment:
I think this method (and related) shouldn't be in this module. I think it rather belongs in app.utils.fba_calculator or something else ; I think we should be clear that this module is all about making calls to cffdrs - and not a place where we do out own calculations.